flow/FileIdentifier.h

```diff
@@ -52,16 +61,27 @@ struct HasFileIdentifier : std::false_type {};
 template <typename T>
 struct HasFileIdentifier<T, decltype((void)FileIdentifierFor<T>::value, 0)> : std::true_type {};
 
-template <class T, uint32_t B, bool = HasFileIdentifier<T>::value>
+template <class T, uint8_t B, bool = (HasFileIdentifier<T>::value && CompositionDepthFor<T>::value < 2)>
 struct ComposedIdentifier;
 
-template <class T, uint32_t B>
-struct ComposedIdentifier<T, B, true>
-{
-	static constexpr FileIdentifier file_identifier = (B << 24) | FileIdentifierFor<T>::value;
+// Manually specified file identifiers must be less than 2^24.
+// The first 8 bits are used to identify the wrapper classes:
+//   The 5th-8th bits represent the inner wrapper class
+//   The 1st-4th bits represent the outer wrapper class
+// Types with more than two level of composition do not get file identifiers.
+template <class T, uint8_t B>
+struct ComposedIdentifier<T, B, true> {
+	static_assert(CompositionDepthFor<T>::value < 2);
```

Comment:
```suggestion
	static_assert(CompositionDepthFor<T>::value < 2, "This type is too nested to be the root of a flatbuffers message. Please make it a member of a new root type instead.");
```

Reply:
I hope this is descriptive enough. Otherwise we could write a longer description in flow/README.md and link to it here.

Reply:
This static assertion is only to check that `ComposedIdentifier` is implemented correctly. `ComposedIdentifier<T,B,true>` never gets instantiated for types with more than two levels of composition, so if you tried to use the file identifier for these types, the compile-time error you would get is that no file identifier exists.

Reply:
Ok. Hopefully people will know to wrap it in a new type with a file identifier when they see that.